syncapi/streams/stream_pdu.go

```diff
@@ -104,8 +128,7 @@ func (p *PDUStreamProvider) IncrementalSync(
 	var stateDeltas []types.StateDelta
 	var joinedRooms []string
 
-	// TODO: use filter provided in request
-	stateFilter := gomatrixserverlib.DefaultStateFilter()
+	stateFilter := req.Filter.Room.State
```

Comment:
Hmmm, this is the line causing all the tests to fail early (`Newly left rooms appear in the leave section of incremental sync`)

And this line looks new from [my code before](https://github.com/matrix-org/dendrite/blob/783fba2d6abff2bfcc7bbf010387b724e8e2a721/syncapi/storage/shared/syncserver.go#L686). This probably needs a better default.

Reply:
Looks like it is `limit: 0` which is causing the problem. Need to better default it